File: frontend/src/types/job.ts
export type SalaryPeriod = "annual" | "hourly";

export type EmploymentType =
  | "full_time"
  | "part_time"
  | "contract"
  | "internship"
  | "other";

export interface Location {
  city: string;
  state: string;
  country: string;
  is_remote: boolean;
}

export interface Salary {
  amount: number;
  currency: string;
  period: SalaryPeriod;
  display: string;
}

export interface Job {
  id: string;
  title: string;
  description: string;
  company: string;
  location: Location;
  salary: Salary;
  employment_type: EmploymentType;
  posting_date: string | null;
  company_type: string;
  language: string;
}

export type SortField = "salary" | "posting_date";
export type SortOrder = "asc" | "desc";

export interface JobSearchParams {
  q?: string;
  country?: string;
  sort_by?: SortField;
  order?: SortOrder;
  page?: number;
  page_size?: number;
}

export interface PaginatedJobs {
  items: Job[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
  has_next: boolean;
  has_prev: boolean;
}

export interface CatalogStats {
  total: number;
  remote: number;
  countries: number;
}

File: backend/app/models/job.py
from __future__ import annotations

from datetime import date
from enum import Enum

from pydantic import BaseModel, Field


class SalaryPeriod(str, Enum):
    ANNUAL = "annual"
    HOURLY = "hourly"


class EmploymentType(str, Enum):
    FULL_TIME = "full_time"
    PART_TIME = "part_time"
    CONTRACT = "contract"
    INTERNSHIP = "internship"
    OTHER = "other"


class Location(BaseModel):
    city: str = ""
    state: str = ""
    country: str = ""
    is_remote: bool = False


class Salary(BaseModel):
    amount: float
    currency: str = "USD"
    period: SalaryPeriod = SalaryPeriod.ANNUAL


class JobPosting(BaseModel):
    id: str
    title: str = ""
    description: str = ""
    company: str = ""
    location: Location = Field(default_factory=Location)
    salary: Salary = Field(default_factory=lambda: Salary(amount=0.0))
    employment_type: EmploymentType = EmploymentType.OTHER
    posting_date: date | None = None
    company_type: str = ""
    language: str = ""
    source_index: int = 0
    source_feed: str = ""


class RejectedJob(BaseModel):
    job: JobPosting
    reasons: list[str]
